lib/slots.ts

```diff
@@ -17,37 +19,41 @@ export type GetSlots = {
 
 const getMinuteOffset = (date: Dayjs, step: number) => {
   // Diffs the current time with the given date and iff same day; (handled by 1440) - return difference; otherwise 0
-  const minuteOffset = Math.min(date.diff(dayjs().startOf("day"), "minutes"), 1440) % 1440;
+  const minuteOffset = Math.min(date.diff(dayjs().startOf("day"), "minute"), 1440) % 1440;
   // round down to nearest step
-  return Math.floor(minuteOffset / step) * step;
+  return Math.ceil(minuteOffset / step) * step;
 };
 
 const getSlots = ({ inviteeDate, frequency, minimumBookingNotice, workingHours }: GetSlots) => {
   // current date in invitee tz
-  const startDate = dayjs(inviteeDate).add(minimumBookingNotice, "minutes"); // + minimum notice period
+  let startDate = dayjs(inviteeDate).add(minimumBookingNotice, "minute");
   // checks if the start date is in the past
   if (startDate.isBefore(dayjs(), "day")) {
     return [];
   }
+  // Add the current time to the startDate if the day is today
+  if (startDate.isToday()) {
```

Comment:
All inviteeDates are exactly that; dates. They don't know about the current time. But if the date is today we do want to take the current time into account.